proposals/p2875.md

```diff
@@ -0,0 +1,467 @@
+arguments.
+
+In the remaining cases, the built-in `Call` interface is used to determine the
+semantics of the call, and the call `F(args)` is translated into
+`F.(Call(ArgTypes).Op)(args)`. Note that this is itself a bound member function
+call, and `Call(ArgTypes)` is a call to a parameterized entity, so this
+transformation is only performed once.
+
+## Details
+
+### Functions and function types
+
+A function declaration such as
+
+```
+fn F[T:! type](x: T) -> T { return x; }
+```
+
+introduces a value named `F`, whose type is a unique _function type_. Distinct
+functions have distinct function types, even if they have the same signature. A
+function type is an empty, trivial type. There is no way to name a function type
+other than asking for the type of the function value.
+
+```
+// Compile-time function.
+fn TypeOf[T:! type](x: T) -> type { return T; }
+
+// ✅ `F` is a first-class value with a first-class type.
+alias FType = TypeOf(F);
+var my_f: FType = F;
+```
+
+Function values are regular values that can be stored in variables, passed to
+functions, and so on.
+
+```
+fn G() -> i32 {
+  // ✅ `my_f` has function type `FType`. This is a direct call to `F`.
+  return my_f(1);
+}
+
+fn Sort[template F:! type, T:! type](v: Vector(T)*, f: F);
+fn Compare(a: i32, b: i32) -> Ordering { return a.(Ordered.Compare)(b); }
+fn SortInts(v: Vector(i32)*) { Sort(v, Compare); }
+```
+
+For the purpose of the orphan rule, a function type is considered to be declared
+by the function declaration that introduces the function value.
+
+### Call syntax
+
+Calls take the form `a(b, c, d)`. This is syntactically equivalent to a primary
+expression followed by a tuple literal, except that no trailing comma is
+required for the case of a single argument, `a(b)`.
+
+The callee can be a name, a literal, a member access, or some more complex
+expression enclosed parentheses.
+
+### Direct calls
+
+When the callee in a call expression can be resolved to a declaration, either
+because it is the name of a parameterized entity like a generic class or
+interface, or because it has a function type, the call is a _direct call_.
+
+In a direct call, a call signature is available which is used to check the given
+arguments against the callee's declared implicit and explicit parameters. This
+checking proceeds as follows:
+
+-   Argument deduction is performed by comparing the declared parameter types
+    against the actual parameter types and deducing values for implicit
+    arguments that make the types equal.
+-   Then, for each binding in the explicit parameter list in turn, all argument
+    values that have been deduced are substituted into the parameter.
+    -   If the parameter is a `template :!` binding, the argument expression is
+        converted to a constant value phase expression of the type of the
+        binding.
+    -   If the parameter is a non-`template` `:!` binding, the argument
```

Comment:
```suggestion
    -   If the parameter is a `:!` binding that is not a `template`, the argument
```

I think this is saying the same as you meant, just suggesting the swap so that the difference of a space not being in backticks isn't significant. This seems particularly subtle for html rendering, but I'm particularly suggesting it because I want to confirm I'm interpreting correctly.